src/roman_datamodels/stnode/_tagged.py

```diff
@@ -45,30 +42,40 @@ def name_from_tag_uri(tag_uri):
     return tag_uri_split
 
 
-class TaggedObjectNode(DNode):
+class _TaggedNodeMixin:
     """
-    Base class for all tagged objects defined by RAD
-        There will be one of these for any tagged object defined by RAD, which has
-        base type: object.
+    Mixin class to provide the common API for all tagged objects.
+
+    Note: the _create_* methods are prefixed with an underscore to prevent them from
+        exposing the builder parameter to as part of the public API. They are the real
+        implementations of the public create_* methods, which simply call the underscored
+        versions without passing a builder. The builder is argument is needed by the Builder
+        objects to pass themselves back through the chain of building calls during nested
+        building operations.
     """
 
-    __slots__ = ()
+    # This is a hack to avoid mypy and __slots__ inheritance issues concerning `_read_tag`
+    #    ideally we would just define `_read_tag` like we did below, but mypy gets upset because
+    #    __slots__ is defined so that the subclasses will be fully slotted. You can't have the
+    #    same slot attributed defined in both parent classes when they are mixed together.
+    if TYPE_CHECKING:
+        __slots__ = ("_read_tag",)
+    else:
+        __slots__ = ()
+
+    _pattern: ClassVar[str]
+    _latest_manifest: ClassVar[str]
 
     _default_tag: ClassVar[str]
+    _read_tag: str
 
-    def __init_subclass__(cls, **kwargs) -> None:
-        """
-        Register any subclasses of this class in the OBJECT_NODE_CLASSES_BY_PATTERN
-        registry.
-        """
-        super().__init_subclass__(**kwargs)
-        if cls.__name__ != "TaggedObjectNode":
-            if cls._pattern in OBJECT_NODE_CLASSES_BY_PATTERN:
-                raise RuntimeError(f"TaggedObjectNode class for tag '{cls._pattern}' has been defined twice")
-            OBJECT_NODE_CLASSES_BY_PATTERN[cls._pattern] = cls
+    def __init__(self, *args, **kwargs):
+        pass
```

Comment:
```suggestion
    def __init__(self, *args, **kwargs):
        pass
```
Is this needed?

Reply:
Yes, MyPy needs it down on lines 80 and 253, so that it knows the "class initializer" accepts inputs. Since this is immediately overloaded and the mixin is never used on its own, this code is only there to make MyPy happy.

Reply:
If there's a solution that involves telling mypy to ignore this I'd prefer that over this empty `__init__`. If not, would you add a comment describing why it needs to be here to appease mypy?

Reply:
I ported part of the changes for my solution to this PR. It is extremely minor.